doc/casync.rst

```diff
@@ -241,3 +246,46 @@ Archive features
 The various ``--with=`` and ``--without=`` parameters control the precise set
 of metadata to store in the archive, or restore when extracting. These flags
 only apply if ``casync`` operates on the file system level.
+
+Excluding Files and Directories from Archiving
+----------------------------------------------
+
+When generating an archive or index from a file system directory tree, some
+files and directories are excluded by default and others may optionally be
+excluded:
+
+1. Files and directories of virtual API file systems exposed by the kernel
+   (i.e. procfs, sysfs, …) are excluded unconditionally.
```

Comment:
hmm, we tar/untar /dev, iirc? The current code does not consider that a "virtual API file system"... It's a special instance of tmpfs after all, and there are no files in there that have strange semantics, everything stored there is pretty regular. All the magic is in the fact that stuff shows up and is removed from there magically...

This is very different from procfs and sysfs where there are tons of really strange files which claim to have size 0 but you can read tons of data from them anyway. Or files that you can only issue ioctls on but never read from (such as /proc/self/ns/*)... That's why we avoid those, since we couldn't even make a sane backup of them if we wanted...

Anyway, I'll slightly extend the documentation here, and explicitly indicate that /dev (and all other tmpfs) are not affected by this implicit exclusion